### ISSUE TITLE
Add Click (Magic Remote) Support

### ISSUE THREAD

**Issue by chiefcll** [lightning-js/renderer#179]
For magic remote users lets add click handling.

**Comment by wouterlucas** [comment]
IMHO there shouldn't be "click" handling in its literal form, the Lightning 3 renderer doesn't directly interact with input devices for key input and neither should it for anything else. This should come from the framework that implements the renderer.

Whether it is touch, a (magic) mouse, spatial input or neuralink the only thing the renderer will need to answer is what node(s) intersect with a set of coordinates. That way, touch, hover, mouse, spatial, etc can all be implemented in a universal way.

@erikhaandrikman you once created a touch library with a bunch of hitbox/intersection logic. We can start with something as basic as L2 and expand it based on the more advanced ones we did in the L2 automotive PoCs.

**Comment by chiefcll** [comment]
Yes - We'd need some API which is "What node is on top at X, Y coordinate" and the framework could ask for that on click events.

**Comment by pecoram** [comment]
Do you have this feature planned?
